feat(security/trivy): print findings table to stdout in addition to SARIF

## Problem / Goal

`actions/security/trivy` runs Trivy with `--format sarif --output /workspace/<file>` only. Findings go to a SARIF file (uploaded to Code Scanning) but **nothing useful lands in stdout**. When the scan fails on `--exit-code 1`, the workflow log shows only:

```text
INFO   [vuln] Vulnerability scanning is enabled
INFO   Detected OS  family="debian" version="13.4"
INFO   Number of language-specific files  num=5
WARN   Using severities from other vendors for some vulnerabilities. ...
##[error]Process completed with exit code 1.
```

The actual finding table — package, CVE, severity, fixed version, justification — is in the SARIF, which is JSON-shaped and intended for the Code Scanning UI, not for a human reading logs at 2am.

Concrete impact today: `wphillipmoore/standard-tooling-docker` PR [#63](https://github.com/wphillipmoore/standard-tooling-docker/pull/63) merged, the post-merge `docker-publish.yml` run failed on all 13 matrix entries + `dev-base` at the Trivy gate, and the on-call view (the workflow log) showed no findings. The triage required pulling the SARIF, or — what actually happened — rebuilding the image locally and re-running Trivy with `--format table` to see what blocked.

## Proposed change

Run Trivy twice (cheaply — DB is cached after the first invocation), or use Trivy's recently-supported multi-format output, so both:

- the SARIF (as today, for Code Scanning upload)
- a `--format table` output (to stdout)

happen in the same step. Optionally include `--scanners vuln --severity CRITICAL,HIGH` consistent with the gate so the human-readable output matches what blocked the run.

Pseudocode for the action's `Run Trivy image scan` step:

```bash
docker run ... trivy image \
  --scanners "$SCANNERS" \
  --severity "$SEVERITY" \
  --exit-code 0 \
  --format table \
  $TRIVYIGNORE_ARGS \
  "$SCAN_REF"

docker run ... trivy image \
  --scanners "$SCANNERS" \
  --severity "$SEVERITY" \
  --exit-code "$EXIT_CODE" \
  --format sarif \
  --output "/workspace/$OUTPUT_FILE" \
  $TRIVYIGNORE_ARGS \
  "$SCAN_REF"
```

The first invocation (table to stdout) sets `--exit-code 0` so it doesn't short-circuit; the second invocation (SARIF) carries the gate's actual `--exit-code` so the gating behavior is unchanged. Order matters: the table prints regardless of pass/fail before the SARIF run can fail the step.

Trivy's DB is downloaded once per container; the second invocation reuses the cached state. Cost is small.

## Acceptance criteria

- [ ] When Trivy finds blocking vulnerabilities, the workflow log shows a human-readable table of CVE + package + installed version + fixed version + severity, in the same step.
- [ ] Existing SARIF upload behavior unchanged — still uploads, still gates on `--exit-code`.
- [ ] Filesystem (`fs`) and image (`image`) scan modes both get the treatment. SBOM mode unaffected.
- [ ] Apply to all three callers in `wphillipmoore/standard-tooling-docker` (`docker-publish.yml` × 2 jobs) and any other repos using `actions/security/trivy@v1.x` — typically a transparent upgrade since the action is invoked the same way.

## Out of scope

- Changing severity thresholds or default exit codes.
- Restructuring the SARIF / Code Scanning integration.
- Adding new scan types.

## Related

- `wphillipmoore/standard-tooling-docker#56` — Trivy-target fix (scan local-built image, not stale `:latest`). Made findings actually surface; this issue makes them readable.
- `wphillipmoore/standard-tooling-docker#64` — current CVE-triage backlog blocked by this exact failure mode (had to reproduce locally to see findings).

--- a/actions/security/trivy/action.yml
+++ b/actions/security/trivy/action.yml
@@ -53,23 +53,56 @@ runs:
     - name: Run Trivy filesystem scan
       if: inputs.scan-type == 'fs'
       shell: bash
+      env:
+        TRIVY_IMAGE: ${{ inputs.trivy-image }}
+        SCAN_REF: ${{ inputs.scan-ref }}
+        SCANNERS: ${{ inputs.scanners }}
+        SEVERITY: ${{ inputs.severity }}
+        EXIT_CODE: ${{ inputs.exit-code }}
+        OUTPUT_FILE: ${{ inputs.output-file }}
+        TRIVYIGNORES: ${{ inputs.trivyignores }}
       run: |
         TRIVYIGNORE_ARGS=""
-        if [ -n "${{ inputs.trivyignores }}" ]; then
-          TRIVYIGNORE_ARGS="--ignorefile ${{ inputs.trivyignores }}"
+        if [ -n "$TRIVYIGNORES" ]; then
+          TRIVYIGNORE_ARGS="--ignorefile $TRIVYIGNORES"
         fi
+        # Single docker session: scan once to JSON, then convert to both
+        # table (stdout, never gates) and SARIF (file, gates per input).
+        # `trivy convert` operates on the cached JSON — no re-scan, no
+        # extra DB download, no extra vuln matching.
         docker run --rm \
           -v "$GITHUB_WORKSPACE:/workspace" \
           -w /workspace \
-          "${{ inputs.trivy-image }}" \
-          fs \
-          --scanners "${{ inputs.scanners }}" \
-          --severity "${{ inputs.severity }}" \
-          --exit-code "${{ inputs.exit-code }}" \
-          --format sarif \
-          --output "/workspace/${{ inputs.output-file }}" \
-          $TRIVYIGNORE_ARGS \
-          "${{ inputs.scan-ref }}"
+          -e SCANNERS \
+          -e SEVERITY \
+          -e EXIT_CODE \
+          -e OUTPUT_FILE \
+          -e SCAN_REF \
+          -e TRIVYIGNORE_ARGS \
+          --entrypoint sh \
+          "$TRIVY_IMAGE" \
+          -c '
+            set -e
+            trivy fs \
+              --scanners "$SCANNERS" \
+              --severity "$SEVERITY" \
+              --exit-code 0 \
+              --format json \
+              --output /tmp/trivy-raw.json \
+              $TRIVYIGNORE_ARGS \
+              "$SCAN_REF"
+            trivy convert \
+              --severity "$SEVERITY" \
+              --format table \
+              --exit-code 0 \
+              /tmp/trivy-raw.json
+            trivy convert \
+              --severity "$SEVERITY" \
+              --format sarif \
+              --exit-code "$EXIT_CODE" \
+              --output "/workspace/$OUTPUT_FILE" \
+              /tmp/trivy-raw.json
+          '
 
     - name: Upload SARIF (filesystem scan)
       if: inputs.scan-type == 'fs' && always()
@@ -81,24 +114,57 @@ runs:
     - name: Run Trivy image scan
       if: inputs.scan-type == 'image'
       shell: bash
+      env:
+        TRIVY_IMAGE: ${{ inputs.trivy-image }}
+        SCAN_REF: ${{ inputs.scan-ref }}
+        SCANNERS: ${{ inputs.scanners }}
+        SEVERITY: ${{ inputs.severity }}
+        EXIT_CODE: ${{ inputs.exit-code }}
+        OUTPUT_FILE: ${{ inputs.output-file }}
+        TRIVYIGNORES: ${{ inputs.trivyignores }}
       run: |
         TRIVYIGNORE_ARGS=""
-        if [ -n "${{ inputs.trivyignores }}" ]; then
-          TRIVYIGNORE_ARGS="--ignorefile ${{ inputs.trivyignores }}"
+        if [ -n "$TRIVYIGNORES" ]; then
+          TRIVYIGNORE_ARGS="--ignorefile $TRIVYIGNORES"
         fi
+        # Single docker session: scan once to JSON, then convert to both
+        # table (stdout, never gates) and SARIF (file, gates per input).
+        # `trivy convert` operates on the cached JSON — no re-scan, no
+        # extra DB download, no extra vuln matching.
         docker run --rm \
           -v "$GITHUB_WORKSPACE:/workspace" \
           -v /var/run/docker.sock:/var/run/docker.sock \
           -w /workspace \
-          "${{ inputs.trivy-image }}" \
-          image \
-          --scanners "${{ inputs.scanners }}" \
-          --severity "${{ inputs.severity }}" \
-          --exit-code "${{ inputs.exit-code }}" \
-          --format sarif \
-          --output "/workspace/${{ inputs.output-file }}" \
-          $TRIVYIGNORE_ARGS \
-          "${{ inputs.scan-ref }}"
+          -e SCANNERS \
+          -e SEVERITY \
+          -e EXIT_CODE \
+          -e OUTPUT_FILE \
+          -e SCAN_REF \
+          -e TRIVYIGNORE_ARGS \
+          --entrypoint sh \
+          "$TRIVY_IMAGE" \
+          -c '
+            set -e
+            trivy image \
+              --scanners "$SCANNERS" \
+              --severity "$SEVERITY" \
+              --exit-code 0 \
+              --format json \
+              --output /tmp/trivy-raw.json \
+              $TRIVYIGNORE_ARGS \
+              "$SCAN_REF"
+            trivy convert \
+              --severity "$SEVERITY" \
+              --format table \
+              --exit-code 0 \
+              /tmp/trivy-raw.json
+            trivy convert \
+              --severity "$SEVERITY" \
+              --format sarif \
+              --exit-code "$EXIT_CODE" \
+              --output "/workspace/$OUTPUT_FILE" \
+              /tmp/trivy-raw.json
+          '
 
     - name: Upload SARIF (image scan)
       if: inputs.scan-type == 'image' && always()

